### DISCUSSION TITLE
[feature request] Multiple expectations in waitUntil()

### DISCUSSION THREAD

**Discussion by d-lord** [Quick/Nimble#902] [Ideas]
When writing tests with multiple asynchronous components (eg using [OHHTTPStubs](https://github.com/AliSoftware/OHHTTPStubs)), it would be nice to do this:

``` swift
waitUntil { clientDone, serverDone in
    stub(alwaysIntercept()) { request in 
        serverDone() // server is satisfied that it received the request
        return OHHTTPStubsResponse(data: someResponse, statusCode: 200, headers: nil)
    }
    MyObj.doWithCallback { _ in
        clientDone() // client is satisfied that it received and handled the response
    }
}
```

At the moment, this doesn't seem to be possible at all, as nested `waitUntil`s are explicitly caught: "Unexpected exception raised: Nested async expectations are not allowed to avoid creating flaky tests."

I'm enjoying using Nimble - it's much cleaner to write with than XCTest - but in this case, XCTest's `expectation`s seem to be the better tool for the job.

**Comment by jeffh** [comment]
Thanks for filing an issue, @d-lord.

I agree, having `waitUntil` or another Nimble function that can specify the number of successful callbacks would be valuable.

**Comment by LuAndreCast** [comment]
any update on this?
🙏🏽
